src/MouseAdapter.js

```diff
@@ -309,20 +383,196 @@ define(function(require) {
+            this.onMouseWheelStart.dispatch([{
+                distance: { x: 0, y: 0 },
+                source: event
+            }]);
+            this._wheeling = true;
+        },
+
+        /**
+         * Dispatches the MouseWheel end event, and handles related controls.
+         * @param {object} event
+         * @private
+         */
+        _dispatchMouseWheelEnd: function(event) {
+
+            // Reset the delta array on MouseEnd to prevent false positives between
+            // natural end and start events.
+            deltaArray = [[],[]];
+
+            this.onMouseWheelEnd.dispatch([{
+                distance: { x: 0, y: 0 },
+                source: event
+            }]);
+            this._wheeling = false;
+        },
+
+
+        /**
+         * Perform a ReScroll dispatch, which currently means dispatch a MouseEnd event 
+         * followed by a MouseStart event, to alert listeners that the user performed a
+         * new distinct scroll event.  This could easily be refactored into a separate
+         * observable if desired in the future.
+         * @param {object} event
+         * @private
+         */
+        _dispatchUserReScroll: function(event) {
             this._dispatchMouseWheelEnd(event);
+            this._dispatchMouseWheelStart(event);
+        },
+
+        /**
+         * Detect when the user performs a new scroll gesture.
+         * @param {object} normalizedEvent
+         * @private
+         */
+        _detectUserScroll: function(normalizedEvent) {
+
+            /*
+             * This is a helper function which takes in a set of consecutive linear data 
+             * and tries to determine whether it is increasing or decreasing.  It can
+             * operate on an entire array, or only a specified section of it. When given a
+             * starting point and a length that extends outside the boundaries of the array,
+             * the function assumes the data wraps around to the beginning of the array.
+             * @param {Array.<number>} array The array of data to perform analysis on.
+             * @param {number} arrayMaxLength [optional] The max length of the array.  If
+             * not given, assumed to be array.length
+             * @param {number} startAt [optional] The index in the array where the target 
+             * data begins.  If not specified, assumes this index is 0.
+             * @param {number} length [optional] The number of data points to examine.  
+             * If not specified, assumed to be the entire array length.
+             * @return {number} Returns:
+                    1 if data appears to be increasing.
+                   -1 if data appears to be decreasing.
+                    0 if there is no discernable trend.
+             */
+            var calculateDeltaDirection = function(array,startAt,length,arrayMaxLength) {
+                arrayMaxLength = arrayMaxLength || array.length;
```

Comment:
The function already (sort of) guards against out of bounds access by modding the array indexes to ensure they are between 0 and max.  (I say sort of because javascript doesn't *really* have bounds.  It wont error, just returns `undefined`).  Also, sometimes out of bounds is expected.  This function looks at X number of data points, but until we have collected at least X to look at some of them are going to be `undefined`.  It handles these cases implicitly, not adjusting `total` in either direction as `undefined <> anything` is false.  This is much simpler than pre-screening the data and/or adjusting the function call with some variable to avoid out of bounds